### PR TITLE
1.0 API Migration Guide

### PR DESCRIPTION
**Pull request by ffuuugor**
Doc what to do if you have opacus code and want to update to the latest version
(Written as a reflection on migrations I did myself)

### CODE REVIEW

File: Migration_Guide.md [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+model = Net()
+optimizer = SGD(model.parameters(), lr=0.05)
+
++ # in addition to model and optimizer you now need access to a data loader
++ data_loader = torch.utils.data.DataLoader(dataset, batch_size=1024)
+
++ # PrivacyEngine now can be initalized with no parameters
++ privacy_engine = PrivacyEngine()
+- privacy_engine = PrivacyEngine(
+
```
Comment (karthikprasad) [inline Migration_Guide.md:81]: this diff looks a bit missed up.

**Review by karthikprasad — APPROVED** [review]
many teeny tiny nits. Completely ignorable.

File: Migration_Guide.md [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+standard PyTorch `DataLoader` and take an example we used to demonstrate old API in our readme.
+
+<pre>
+model = Net()
+optimizer = SGD(model.parameters(), lr=0.05)
+
+<span style="color: #a6a6a6"># in addition to model and optimizer you now need access to a data loader</span>
+<span style="background-color: #aef2b5">data_loader = torch.utils.data.DataLoader(dataset, batch_size=1024)</span> 
+
+<span style="color: #a6a6a6"># PrivacyEngine now can be initalized with no parameters</span>
```
[the remark this answers is not in the document]
Comment (ffuuugor) [reply Migration_Guide.md]: Not particularly true - there's `secure_mode` and `accountant`.
Maybe "PrivacyEngine's constructor doesn't accept training artefacts - they're instead passed to make_private"

File: Migration_Guide.md [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+## No DataLoader
+
+Now, if you're using something else as your data source, things get interesting. You're still be able to use Opacus,
+but will need to do a little more.
+
+`PrivacyEngine` is intentionally designed to expect and amend `DataLoader`, as this is the right thing to in the 
+majority of cases. However, the good news is that `PrivacyEngine` itself is not absolutely necessary - if you
+know what you're doing, and are happy with whatever data source you have, here's how to plug in opacus.
+
+NB: This is only a brief example of using Opacus components independently of `PrivacyEngine`.
```
[the remark this answers is not in the document]
Comment (ffuuugor) [reply Migration_Guide.md:215]: https://en.wikipedia.org/wiki/Nota_bene

Don't know where I picked this up, but urban dictionary knows about it!

[the remark this answers is not in the document]
Comment (karthikprasad) [reply Migration_Guide.md:214]: > but urban dictionary knows about it!

That settles it then! Stamp!

**Pull request merged**